### PR TITLE
Dual role#15

### PR DESCRIPTION
**Pull request by superolegatron**
We have already had the major part of dual role here are new addings:

- Users
Now we have endpoint `api/users/profile`. This path returns info about current authenticated user on GET, also we can edit on PUT and soft delete on DELETE, but if you want to delete user you have to have no investir and strtup profiles

- Investors
Also adding of `api/investors/profile` to have an endpoint directly to icurrent nvestor profile if user is investor

- Startups
Also adding of `api/startups/profile` to have an endpoint directly to current startup profile if user is startup 
Added soft delete, but you can only delete a startup if you have no projects
Added permissions classes for StartupViewSet

- Projects
Added soft delete 

If there any missing functionality for dual role - looking forward to your feedback

### CODE REVIEW

**Review by NovitskaMariia — COMMENTED** [review]
I think, user wants to have an opportunity to update just one or a few fields. When I tried to update user's profile with just one field, I couldn't. We don't need both PUT and PATCH methods, so, here we can replace PUT --> PATCH
image

File: users/views.py [hunk 1]
```diff
@@ -215,3 +218,43 @@ def post(self, request, uidb64, token):
...
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def delete(self, request):
+        user_instance = self.get_current_user(request)
+
+        if user_instance.is_investor == 1:
```
Comment (VovaD199) [inline users/views.py:250]: We can have only one startup, but there can be several investors. Please make the appropriate changes.

**Review by NovitskaMariia — APPROVED** [review]

File: users/views.py (hunk 1 above)
Comment (superolegatron) [reply users/views.py:250]: We have one startup and one investor per user

**Comment by Thoughtseize1** [comment]
Using `/api/startups/profile` maybe you can add  `industries = serializers.CharField(source='industries.name')` to class `StartupSerializer`.

If we want to receive the title of industry, rather than an ID.

**Review by Thoughtseize1 — APPROVED** [review]

**Pull request merged**